core/kazoo_modb-1.0.0/src/kazoo_modb.erl

```diff
@@ -286,3 +287,26 @@ should_archive(AccountMODb, Year, Month) ->
             ModbMonths = (ModbYear * 12) + ModbMonth,
             (Months - ModbMonths) > whapps_config:get_integer(?CONFIG_CAT, <<"active_modbs">>, 6)
     end.
+
+
+%%--------------------------------------------------------------------
+%% @doc
+%% Delete an modb if it is no longer associated with its account.
+%% (That is: orphaned).
+%% AccountMODb must be 'encoded' otherwise couch_mgr:db_delete/1 will
+%% fail. Returns whether AccountMODb has been deleted.
+%% @end
+%%--------------------------------------------------------------------
+-spec maybe_delete (ne_binary()) -> boolean().
+maybe_delete (AccountMODb) ->
+    AccountId = wh_util:format_account_id(AccountMODb, 'raw'),
+    AccountIds = whapps_util:get_all_accounts('raw'),
```

Comment:
I wonder if you could make a `maybe_delete/2` that takes the list of AccountIds, so that the calling function, when it knows it will be doing a lot of operations like this, could pass the list in instead. Consider when we have 10,000+ databases, calculating AccountIds for a large number of MODbs would put a bit of strain on the DB cluster.